src/com_tjnotifications/admin/controllers/notifications.php

```diff
@@ -56,12 +56,18 @@ public function delete()
 		Session::checkToken() or jexit(Text::_('JINVALID_TOKEN'));
 
 		$mainframe = Factory::getApplication();
-		$extension = $mainframe->input->get('extension', '', 'STRING');
-		$cid       = $mainframe->input->get('cid', array(), 'array');
+		$input     = $mainframe->input;
+		$extension = $input->get('extension', '', 'STRING');
+		$cid       = $input->get('cid', array(), 'array');
+		$recordId  = (int) (count($cid) ? $cid[0] : $input->getInt('id'));
```

Comment:
does batch delete work in this case? If not change code to support that.
Again, you can generate code for same using this https://github.com/manojLondhe/joomla-extensions-generator